src/main/java/org/jvnet/hudson/update_center/HPI.java

```diff
@@ -110,6 +112,17 @@ public String getCompatibleSinceVersion() throws IOException {
         return getManifestAttributes().getValue("Compatible-Since-Version");
     }
 
+    public VersionNumber getMinimumJavaVersion() throws IOException {
+        String manifestEntry = getManifestAttributes().getValue("Minimum-Java-Version");
+        if (StringUtils.isNotBlank(manifestEntry)) {
+            return new VersionNumber(manifestEntry);
+        }
+
+        //TODO(oleg_nenashev): It will break plugins which intentionally declare lower version than the one required by the core (which is a bad idea)
```

Comment:
We discussed it with @daniel-beck and other stakeholders and agreed that we will disable it for now so that we do not enforce checks for older plugin versions.